opendatahub-io/data-science-pipelines-operator
# Chore: Remove v1 fields and reconcile logic from dspo

## The issue resolved by this Pull Request:
Resolves #https://issues.redhat.com/browse/RHOAIENG-13572

## Description of your changes:

This PR removes V1 from DSPO, more specifically it: 

- [x] Removes all v1 only fields from DSPA CR
- [x] DSPA version field [here](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/samples/v2/dspa-all-fields/dspa_all_fields.yaml#L12), is assumed as "v2" by default when not provided (previously it was v1)
	- [x] If user specifies v1, DSPO throws a reconcile log message and does not proceed with the reconcilliation for that 
	- [x] If user specifies v1, DSPA status reports a meaningful error with suggested action
- [x] DSPO continues to reconcile on the v1alpha1 resources (for both v1/v2), but when it encounters v1 version set in .spec.dspversion field in DSPA, dspo logs an error and dspa status error condition is thrown
- [x] DSPA CRD continues to serve at v1apha1, but has a deprecation set with a warning
- [x] DSPA uses a conversion strategy of None (default) [1](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#writing-reading-and-updating-versioned-customresourcedefinition-objects). In this scenario when fetching the v1alpha1 resources, they are served with the unknown v1 fields as pruned.
- [x] Removes tekton related RBAC grants to pipeline runner SA, as well as the same rbac requirements on the DSPO
- [x] Removes any logic that will reconcile on v1 related fields
- [x] Updates all unit tests and functional tests to only test for v2, and remove any v1 related tests
- [x] DSPO should not trigger reconcilliations on custom watches on resources based on labels that belong to v1, example [this](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/controllers/dspipeline_controller.go#L624)


## Testing instructions

DSPA v2 before after upgrade diff, only showing the relevant bits, the rest is the same
![image](https://github.com/user-attachments/assets/da80c98a-c338-4f12-82a8-b4a80b316262)

When v1 is detected, DSPO will log: 

```txt
2024-10-01T20:06:08Z INFO unsupported DSP version v1 detected. Please manually remove this DSP resource and re-apply with a supported version field set {"namespace": "dspa1", "dspa_name": "sample"}
```

and show the following dspa status: 

![image](https://github.com/user-attachments/assets/3cbafb20-f965-49cb-be14-995af8912528)

otherwise the DSPA v1 cr is left untouched, only the removed deprecated fields are synch'd by the cluster. 


DSPO logs should not be showing any stacktrace errors.
